### PR TITLE
convert Log4j mdc test from groovy to java

### PR DESCRIPTION
**Pull request by niteshs7**
Related to #7195

### CODE REVIEW

File: instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/log4j/mdc/v1_2/Log4j1MdcTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+public class Log4j1MdcTest {
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static final Logger logger = LogManager.getLogger("TestLogger");
+
+  @BeforeEach
+  public void cleanup() {
```
Comment (mateuszrzeszutek) [inline instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/log4j/mdc/v1_2/Log4j1MdcTest.java:30]: Nit: Junit5 classes & methods don't need to be public:
```suggestion
  void cleanup() {
```

File: instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/log4j/mdc/v1_2/ListAppender.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    events.clear();
+  }
+
+  public static ArrayList<LoggingEvent> getEvents() {
+    return events;
+  }
+
+  public static void setEvents(ArrayList<LoggingEvent> events) {
+    ListAppender.events = events;
+  }
```
[the remark this answers is not in the document]
Comment (niteshs7) [reply instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/log4j/mdc/v1_2/ListAppender.java]: Done.

File: instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/log4j/mdc/v1_2/Log4j1MdcTest.java (hunk 1 above)
Comment (niteshs7) [reply instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/log4j/mdc/v1_2/Log4j1MdcTest.java]: Done. Thanks for the review Mateusz !

**Review by laurit — APPROVED** [review]

File: instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/log4j/mdc/v1_2/ListAppender.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+package io.opentelemetry.javaagent.instrumentation.log4j.mdc.v1_2;
+
+import java.util.ArrayList;
+import org.apache.log4j.AppenderSkeleton;
+import org.apache.log4j.spi.LoggingEvent;
+
+public class ListAppender extends AppenderSkeleton {
+
+  private static final ArrayList<LoggingEvent> events = new ArrayList<LoggingEvent>();
```
[the remark this answers is not in the document]
Comment (niteshs7) [reply instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/log4j/mdc/v1_2/ListAppender.java]: Done. Thanks Lauri!

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**